### PR TITLE
fix: optional port for nod schema

### PR DESCRIPTION
**Pull request by theweakgod**
### Description

there is a bug in the schema check of nodes when admin-api saves the upstream configuration.

Fixes # ([issue](https://github.com/apache/apisix/issues/10455))

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by yujiapingyu** [comment]
cheers

**Comment by monkeyDluffy6017** [comment]
@theweakgod Could you add a test case to cover this?

**Comment by theweakgod** [comment]
> @theweakgod Could you add a test case to cover this?

Ok, can you tell me how to add a test case. Or some doc that tells me how to add it

**Comment by monkeyDluffy6017** [comment]
@theweakgod you could refer to https://apisix.apache.org/blog/2022/06/27/getting-start-with-apisix-test-cases/

**Comment by theweakgod** [comment]
> https://apisix.apache.org/blog/2022/06/27/getting-start-with-apisix-test-cases/

thanks

**Comment by theweakgod** [comment]
> @theweakgod Could you add a test case to cover this?

done!

**Comment by shreemaan-abhishek** [comment]
@theweakgod, the test case you added aren't up to the mark. Please demonstrate using domain name in the nodes without port is acceptable and accessing a route that uses such upstream does not fail.

File: t/core/schema_def_optional_port.t [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+__DATA__
+
+=== TEST 1: Sanity check node_schema optional port
+--- config
+    location /t {
+        content_by_lua_block {
+            local schema_def = require("apisix.schema_def")
+            local core = require("apisix.core")
+
+
```
Comment (monkeyDluffy6017) [inline t/core/schema_def_optional_port.t:47]: redundant blank

File: t/core/schema_def_optional_port.t
Comment (shreemaan-abhishek) [inline t/core/schema_def_optional_port.t:1]: @theweakgod, the test case you added aren't up to the mark. Please demonstrate using domain name in the nodes without port is acceptable and accessing a route that uses such upstream does not fail.

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]

File: t/core/schema_def_optional_port.t
Comment (monkeyDluffy6017) [inline t/core/schema_def_optional_port.t:1]: No need to add a new test file, add it here: https://github.com/apache/apisix/blob/master/t/admin/schema-validate.t

Comment (monkeyDluffy6017) [reply t/core/schema_def_optional_port.t:1]: Yeah, you should prove that the route could work

**Comment by theweakgod** [comment]
@monkeyDluffy6017 @shreemaan-abhishek all done

**Comment by monkeyDluffy6017** [comment]
please make the ci pass

**Comment by theweakgod** [comment]
> 请让 ci 通过

ok，updated

**Comment by theweakgod** [comment]
> please make the ci pass

Is this error a problem with the test file? What do I need to do?

**Comment by Revolyssup** [comment]
@theweakgod It was a flaky test. I rerun the CI and it passed.

**Comment by theweakgod** [comment]
> @theweakgod It was a flaky test. I rerun the CI and it passed.

Ok, I see.

File: t/admin/schema-validate.t [hunk 2]
```diff
@@ -398,3 +398,44 @@ location /t {
...
+--- request
+GET /t
+--- response_body
+passed
+
+
+
+=== TEST 14: Test route upstream
+--- request
+GET /ip
```
Comment (monkeyDluffy6017) [inline t/admin/schema-validate.t:441]: @Revolyssup Would you like to update these test cases? it is better not to use `httpbin.org`, it is unstable, and we could check the response body like the other test cases, and no need to add `X-API-KEY` in the request header and so on

Comment (Revolyssup) [reply t/admin/schema-validate.t:441]: okay

Comment (theweakgod) [reply t/admin/schema-validate.t:439]: > okay

I remove some useless code.

File: t/admin/schema-validate.t [hunk 3]
```diff
@@ -398,3 +398,42 @@ location /t {
...
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes',
+                ngx.HTTP_POST,
+                {
+                    uri = "/ip",
+                    upstream = {
+                        type = "roundrobin",
+                        nodes = {
+                            { host = "nghttp2.org", weight = 1,}
```
Comment (monkeyDluffy6017) [inline t/admin/schema-validate.t:416]: use `127.0.0.1:1980` instead

File: t/admin/schema-validate.t [hunk 4]
```diff
@@ -398,3 +398,42 @@ location /t {
...
+
+=== TEST 13: Check node_schema optional port
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes',
+                ngx.HTTP_POST,
+                {
+                    uri = "/ip",
```
Comment (monkeyDluffy6017) [inline t/admin/schema-validate.t:412]: you could search `127.0.0.1:1980` in the project, and find what uri is better

Comment (theweakgod) [reply t/admin/schema-validate.t]: OK, done. thanks

File: t/admin/schema-validate.t [hunk 5]
```diff
@@ -398,3 +398,42 @@ location /t {
...
+--- request
+GET /t
+--- response_body
+passed
+
+
+
+=== TEST 14: Test route upstream
+--- request
+GET /ip
```
Comment (monkeyDluffy6017) [inline t/admin/schema-validate.t]: We should check the response body, like this
```
=== TEST 8: `url` exist and `route_id` is 1
--- request
GET /hello
--- response_body
hello world
```

**Comment by monkeyDluffy6017** [comment]
@theweakgod Nice job!

**Comment by theweakgod** [comment]
> @theweakgod Nice job!

^ ^. thank you very much for your help.

**Review by Revolyssup — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Pull request merged**